schema/schema_validator.py

```diff
@@ -165,17 +171,24 @@ def validate_test_data_with_schema(self):
             all_results.append(result_data)
         return all_results
 
-    def parallel_check_test_data_schema(self, schema_test_data):
-        num_processors = mp.cpu_count()
-        logging.info('Schema validation: %s processors for %s schema/test data pairs',
-                     num_processors,
-                     len(schema_test_data))
-
-        # Returns all the results
-        processor_pool = mp.Pool(num_processors)
-        with processor_pool as p:
-            result = p.map(self.check_test_data_against_schema, schema_test_data)
-        return result
+    def check_all_test_data_schema(self, schema_test_data):
+        if not self.run_serial:
+            num_processors = mp.cpu_count()
+            logging.info('test data parallel validation: %s processors for %s schema/test data pairs',
+                         num_processors,
+                         len(schema_test_data))
+
+            # Returns all the results
+            processor_pool = mp.Pool(num_processors)
+            with processor_pool as p:
+                result = p.map(self.check_test_data_against_schema, schema_test_data)
+            return result
+        else:
+            results = []
+            logging.info('test data running serially on %s files!', len(schema_test_data))
+            for test_data in schema_test_data:
+                results.append(self.check_test_data_against_schema(test_data))
```

Comment:
could turn the instantiation + initialization of `results` into a FP-style `map`. Not sure how Python does that.